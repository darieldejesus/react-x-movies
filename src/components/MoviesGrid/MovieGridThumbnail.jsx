import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'styled-components';

const Container = Styled.div`
  display: block;
  width: 154px;
`;

const Image = Styled.img`
  background: red;
  height: 231px;
  width: 154px;
`;

const MovieName = Styled.a`
  color #63676f;
  font-size: 1rem;
  font-weight: 300;
  margin: 10px 0 0;
  text-decoration: none;
  text-transform: uppercase;
`;

const MovieReleaseDate = Styled.h4`
  color: #ed0005;
  font-size: 0.8rem;
  font-weight: 600;
  margin: 0;
`;

const MovieGridThumbnail = ({
  id,
  title,
  poster_path,
  release_date,
  vote_average,
  vote_count,
  popularity,
}) => (
  <Container>
    <Image src={`https://image.tmdb.org/t/p/w154${poster_path}`} />
    <MovieName href={`/movie/${id}`}>{title}</MovieName>
    <MovieReleaseDate>{popularity} </MovieReleaseDate>
    <MovieReleaseDate>{vote_average} ({vote_count})</MovieReleaseDate>
    <MovieReleaseDate>({release_date})</MovieReleaseDate>
  </Container>
);

MovieGridThumbnail.propTypes = {
  id: PropTypes.number.isRequired,
  popularity: PropTypes.number.isRequired,
  poster_path: PropTypes.string.isRequired,
  release_date: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  vote_average: PropTypes.number.isRequired,
  vote_count: PropTypes.number.isRequired,
};

export default MovieGridThumbnail;
