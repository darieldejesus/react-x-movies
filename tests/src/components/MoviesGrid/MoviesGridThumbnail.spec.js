import React from 'react';
import MovieGridThumbnail from '../../../../src/components/MoviesGrid/MovieGridThumbnail';
import fixture from '../../../fixture/moviesDetails.json';

describe('MoviesGridThumbnail', function () {
  beforeEach(() => {
    this.wrapper = shallow(<MovieGridThumbnail {...fixture} />);
  });

  test('should renders the MoviesGridThumbnail', () => {
    expect(toJson(this.wrapper)).toMatchSnapshot();
  });
});
