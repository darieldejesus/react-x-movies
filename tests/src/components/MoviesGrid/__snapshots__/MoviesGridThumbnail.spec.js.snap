// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesGridThumbnail should renders the MoviesGridThumbnail 1`] = `
<styled.div>
  <styled.img
    src="https://image.tmdb.org/t/p/w154/9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
  />
  <styled.a
    href="/movie/346364"
  >
    It
  </styled.a>
  <styled.h4>
    729.612854
     
  </styled.h4>
  <styled.h4>
    7.3
     (
    3559
    )
  </styled.h4>
  <styled.h4>
    (
    2017-09-05
    )
  </styled.h4>
</styled.div>
`;
